extension fails with yarn2

<!-- 
SEE THIS GUIDE ON OPENING A GOOD ISSUE: https://github.com/prettier/prettier-vscode/wiki/Writing-a-Good-Issue

**You must complete all steps in this checklist or the issue will be automatically closed.**

- [x ] Write a summary of your issue
- [ ] Include a link to a github repo that can be used to reproduce the issue or exact reproduction steps.
- [ ] Provide the expected result
- [ x] Provide the actual result
- [ x] Include extension, os, and vs code information
- [x ] Include Prettier extension log

PLEASE NOTE: Many of the issues with the extension are heavily based on your specific configuration and source code. In order to improve your chances of having your bug fixed, the number one thing you can do is create a github repository that can be used to reproduce the issue along with the detailed reproduction steps. If you don't provide this info, it is usually the case that we cannot reproduce the issue and the issue will be closed. 
-->

### Summary

I use yarn2 to install dependencies

### Github Repository to Reproduce Issue

repo is private

### Steps To Reproduce:

1.  setup yarn2
2.  setup all dependencies for the project `yarn install`
3. open any file of the project in vs code with prettier extension

### Expected result

usual behaviour as it was on prior version of yarn

### Actual result

Error: Failed to load module. If you have prettier or plugins referenced in package.json, ensure you have run `npm install` Attempted to load prettier from package.json

Sourse: Prettier - Code Formatter [extension]

![image](https://user-images.githubusercontent.com/8055157/73140934-62c80c80-408f-11ea-83f1-16f1b59d31a1.png)

### Additional information

**VS Code Version**: 1.41.1

**Prettier Extension Version**: 3.18.0

**OS and version**: MacOS 10.15.2 (19C57)

## Prettier Log Output

```
["ERROR" - 11:02:41 PM] Failed to load prettier from ''**Censored**
Cannot find module 'prettier' from '/Users/**Censored**/projects/**Censored**/**Censored**/package.json'
Error: Cannot find module 'prettier' from '/Users/**Censored**/projects/**Censored**/**Censored**/package.json'
	at Function.e.exports [as sync] (/Users/**Censored**/.vscode/extensions/esbenp.prettier-vscode-3.18.0/dist/extension.js:1:319843)
	at t.ModuleResolver.findPkg (/Users/**Censored**/.vscode/extensions/esbenp.prettier-vscode-3.18.0/dist/extension.js:1:246852)
	at t.ModuleResolver.findPkg [as findPkgMem] (/Users/**Censored**/.vscode/extensions/esbenp.prettier-vscode-3.18.0/dist/extension.js:1:247210)
	at t.ModuleResolver.requireLocalPkg (/Users/**Censored**/.vscode/extensions/esbenp.prettier-vscode-3.18.0/dist/extension.js:1:245330)
	at t.ModuleResolver.getPrettierInstance (/Users/**Censored**/.vscode/extensions/esbenp.prettier-vscode-3.18.0/dist/extension.js:1:243969)
	at t.LanguageResolver.getSupportLanguages (/Users/**Censored**/.vscode/extensions/esbenp.prettier-vscode-3.18.0/dist/extension.js:1:242420)
	at t.LanguageResolver.allEnabledLanguages (/Users/**Censored**/.vscode/extensions/esbenp.prettier-vscode-3.18.0/dist/extension.js:1:242028)
	at t.StatusBarService.toggleStatusBarItem (/Users/**Censored**/.vscode/extensions/esbenp.prettier-vscode-3.18.0/dist/extension.js:1:85577)
	at /Users/**Censored**/.vscode/extensions/esbenp.prettier-vscode-3.18.0/dist/extension.js:1:85250
	at l.fire (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:46:893)
	at /Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:641:245
	at l.fire (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:46:893)
	at f.$acceptDocumentsAndEditorsDelta (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:561:804)
	at p._doInvokeHandler (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:663:275)
	at p._invokeHandler (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:662:971)
	at p._receiveRequest (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:661:639)
	at p._receiveOneMessage (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:660:468)
	at /Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:658:691
	at l.fire (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:46:893)
	at _.fire (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:189:274)
	at /Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:820:285
	at l.fire (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:46:893)
	at _.fire (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:189:274)
	at t.PersistentProtocol._receiveMessage (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:193:629)
	at /Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:190:824
	at l.fire (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:46:893)
	at p.acceptChunk (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:186:737)
	at /Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:186:89
	at Socket.t (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:195:68)
	at Socket.emit (events.js:200:13)
	at addChunk (_stream_readable.js:294:12)
	at readableAddChunk (_stream_readable.js:275:11)
	at Socket.Readable.push (_stream_readable.js:210:10)
	at Pipe.onStreamRead (internal/stream_base_commons.js:166:17)

```

@ntotten the problem here is that even when setting the vscode setting ;

```
"prettier.prettierPath": ".vscode/pnpify/prettier/index.js"`
```

(As done automatically by following https://yarnpkg.com/advanced/editor-sdks )

It looks like the plugins still tries to load the version from `package.json` first instead of the explicitly configured path.

```
["INFO" - 7:15:59 PM] Formatting /Users/xxxx/Code/sterblue2/packages/algorithms/random/src/index.ts
["INFO" - 7:15:59 PM] Using ignore file (if present) at /Users/xxx/Code/sterblue2/.prettierignore
["ERROR" - 7:15:59 PM] Failed to load local module prettier.
["ERROR" - 7:15:59 PM] Cannot find module 'prettier' from '/Users/xxxx/Code/sterblue2/package.json'
Error: Cannot find module 'prettier' from '/Users/xxx/Code/sterblue2/package.json'
	at Function.e.exports [as sync] (/Users/xxx/.vscode/extensions/esbenp.prettier-vscode-3.20.0/dist/extension.js:1:320528)
	at t.ModuleResolver.findPkg (/Users/xxx/.vscode/extensions/esbenp.prettier-vscode-3.20.0/dist/extension.js:1:247279)
	at t.ModuleResolver.findPkg (/Users/xxx/.vscode/extensions/esbenp.prettier-vscode-3.20.0/dist/extension.js:1:247381)
	at t.ModuleResolver.findPkg [as findPkgMem] (/Users/xxxx/.vscode/extensions/esbenp.prettier-vscode-3.20.0/dist/extension.js:1:247637)
	at t.ModuleResolver.requireLocalPkg (/Users/xxxx/.vscode/extensions/esbenp.prettier-vscode-3.20.0/dist/extension.js:1:245697)
	at t.ModuleResolver.getPrettierInstance (/Users/xxxx/.vscode/extensions/esbenp.prettier-vscode-3.20.0/dist/extension.js:1:244301)
	at t.default.<anonymous> (/Users/xxxx/.vscode/extensions/esbenp.prettier-vscode-3.20.0/dist/extension.js:1:348961)
	at Generator.next (<anonymous>)
	at s (/Users/xxxxx/.vscode/extensions/esbenp.prettier-vscode-3.20.0/dist/extension.js:1:345019)
["INFO" - 7:15:59 PM] Using bundled version of prettier.
["INFO" - 7:15:59 PM] File Info:
{
  "ignored": false,
  "inferredParser": "typescript"
}
["INFO" - 7:15:59 PM] Detected local configuration (i.e. .prettierrc or .editorconfig), VS Code configuration will not be used
["INFO" - 7:15:59 PM] Prettier Options:
{
  "filepath": "/Users/xxxx/xxx/index.ts",
  "parser": "typescript",
  "useTabs": false,
  "tabWidth": 2,
  "printWidth": 80,
  "semi": true,
  "singleQuote": false,
  "jsxSingleQuote": false,
  "trailingComma": "none",
  "bracketSpacing": true,
  "jsxBracketSameLine": false,
  "arrowParens": "avoid",
  "htmlWhitespaceSensitivity": "css",
  "endOfLine": "lf"
}
["INFO" - 7:15:59 PM] Formatting completed in 40.413031ms.
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -4,6 +4,10 @@ All notable changes to the "prettier-vscode" extension will be documented in thi
 
 <!-- Check [Keep a Changelog](http://keepachangelog.com/) for recommendations on how to structure this file. -->
 
+## [next]
+
+- Honor project-scoped config when loading module
+
 ## [4.0.0]
 
 - Updated prettier to 2.0 - [See changes here](https://prettier.io/blog/2020/03/21/2.0.0.html).

--- a/src/ModuleResolver.ts
+++ b/src/ModuleResolver.ts
@@ -7,7 +7,7 @@ import * as readPkgUp from "read-pkg-up";
 import * as resolve from "resolve";
 import * as semver from "semver";
 // tslint:disable-next-line: no-implicit-dependencies
-import { Disposable } from "vscode";
+import { Disposable, Uri } from "vscode";
 import { resolveGlobalNodePath, resolveGlobalYarnPath } from "./Files";
 import { LoggingService } from "./LoggingService";
 import { FAILED_TO_LOAD_MODULE_MESSAGE } from "./message";
@@ -88,7 +88,9 @@ export class ModuleResolver implements Disposable {
       return prettier;
     }
 
-    const { prettierPath, packageManager, resolveGlobalModules } = getConfig();
+    const { prettierPath, packageManager, resolveGlobalModules } = getConfig(
+      Uri.file(fileName)
+    );
 
     // tslint:disable-next-line: prefer-const
     let { moduleInstance, modulePath } = this.requireLocalPkg<PrettierModule>(

